jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithModifVarsQueryMapper.java

```diff
@@ -0,0 +1,122 @@
+
+    /**
+     * Dedicated for ServiceLoader to create instance, use <code>get()</code> method instead 
+     */
+    public UserTaskInstanceWithModifVarsQueryMapper() {
+        super();
+    }
+    
+    public static UserTaskInstanceWithModifVarsQueryMapper get() {
+        return new UserTaskInstanceWithModifVarsQueryMapper();
+    }
+
+    @Override
+    public List<UserTaskInstanceWithPotOwnerDesc> map(Object result) {
+        if (result instanceof DataSet) {
+            DataSet dataSetResult = (DataSet) result;
+            List<UserTaskInstanceWithPotOwnerDesc> mappedResult = new ArrayList<UserTaskInstanceWithPotOwnerDesc>();
+                                                           
+            if (dataSetResult != null) {
+                Map<Long, UserTaskInstanceWithPotOwnerDesc> tmp = new HashMap<Long, UserTaskInstanceWithPotOwnerDesc>();
+                
+                for (int i = 0; i < dataSetResult.getRowCount(); i++) {
+                    Long taskId = getColumnLongValue(dataSetResult, COLUMN_TASKID, i);
+                    UserTaskInstanceWithPotOwnerDesc ut = tmp.get(taskId);
+                    if (ut == null) {
+                        ut = buildInstance(dataSetResult, i);
+                        mappedResult.add(ut);    
+                        
+                        tmp.put(taskId, ut);
+                    }else if(!tmp.get(taskId).getPotentialOwners().contains(getColumnStringValue(dataSetResult, COLUMN_POTOWNER, i))){
+                        
+                        ((org.jbpm.kie.services.impl.model.UserTaskInstanceWithPotOwnerDesc)ut).addPotOwner(getColumnStringValue(dataSetResult, COLUMN_POTOWNER, i));
+                    }
+                     
+                    String varName = getColumnStringValue(dataSetResult, COLUMN_TASK_VAR_NAME, i);
+                    String varValue = getColumnStringValue(dataSetResult, COLUMN_TASK_VAR_VALUE, i);
+                    
+                    if(getColumnIntValue(dataSetResult, COLUMN_TASK_VAR_TYPE, i) == 0) {
+                        ((org.jbpm.kie.services.impl.model.UserTaskInstanceWithPotOwnerDesc)ut).addInputdata(varName, varValue);
+                    }else {
+                        ((org.jbpm.kie.services.impl.model.UserTaskInstanceWithPotOwnerDesc)ut).addOutputdata(varName, varValue);
+                    }             
+                }
+            }
+            
+            return mappedResult;
+        }
+        
+        throw new IllegalArgumentException("Unsupported result for mapping " + result);
+    }
+
+    @Override
+    public String getName() {
+        return "UserTasksWithModifications";
+    }
+
+    @Override
+    public Class<?> getType() {
+        return UserTaskInstanceWithPotOwnerDesc.class;
+    }
+
+    @Override
+    public QueryResultMapper<List<UserTaskInstanceWithPotOwnerDesc>> forColumnMapping(Map<String, String> columnMapping) {
+        return new UserTaskInstanceWithModifVarsQueryMapper();
```

Comment:
is it intended to ignore columnMapping?

Reply:
No, why?

Reply:
the method has argument columnMapping but it is never used like it most likely should be added as constructor of return new UserTaskInstanceWithModifVarsQueryMapper()

Reply:
Sorry Maciej, I didn't understand you well. I think is good ignoring the argument columnMapping for this mapper. Similar to: https://github.com/kiegroup/jbpm/blob/master/jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithVarsQueryMapper.java

Reply:
that's why I asked if that was intended or not. If you believe there will not be a need to read up custom variables then it's ok with me.

Reply:
Ok, it's ok